backend/indexer/indexer.py

```diff
@@ -364,8 +365,9 @@ async def ingest_data(request: IngestDataToCollectionDto):
                         raise_error_on_failure=created_data_ingestion_run.raise_error_on_failure,
                         batch_size=request.batch_size,
                     )
-                )
-                created_data_ingestion_run.status = DataIngestionRunStatus.COMPLETED
+                # future of this submission is ignored, ingestion failure due to process termination will not be tracked
+                process_pool.submit(sync_data_source_to_collection, ingestion_config)
```

Comment:
This is almost what I was going for!


Here is what I want to get to https://gist.github.com/chiragjn/882d43a83583a9729a61ee25e2b60c5c

I suppose we will have to wrap `sync_data_source_to_collection` in `asyncio.run` on the process worker to run it, otherwise it won't schedule

Reply:
Ah I see you ended up doing this wrapping in the custom pool implementation, nice

Reply:
Let's just pass the `process_pool` as optional arg